pkg/orchestrator/provider/aws/scanestimation/estimate.go

```diff
@@ -0,0 +1,386 @@
+		sourceSnapshotMonthlyCost, err = s.priceFetcher.GetSnapshotMonthlyCostPerGB(ctx, sourceRegion)
+		if err != nil {
+			return nil, fmt.Errorf("failed to get source snapshot monthly cost: %v", err)
+		}
+	}
+
+	// Fetch the snapshot monthly cost.
+	destSnapshotMonthlyCost, err := s.priceFetcher.GetSnapshotMonthlyCostPerGB(ctx, destRegion)
+	if err != nil {
+		return nil, fmt.Errorf("failed to get dest snapshot monthly cost: %v", err)
+	}
+
+	// Fetch the scanner job monthly cost.
+	scannerPerHourCost, err := s.priceFetcher.GetInstancePerHourCost(ctx, destRegion, scannerInstanceType, marketOption)
+	if err != nil {
+		return nil, fmt.Errorf("failed to get scanner per hour cost: %v", err)
+	}
+
+	// Fetch the scanner job volume monthly cost.
+	scannerRootVolumeMonthlyCost, err := s.priceFetcher.GetVolumeMonthlyCostPerGB(ctx, destRegion, scannerVolumeType)
+	if err != nil {
+		return nil, fmt.Errorf("failed to get volume monthly cost per GB: %v", err)
+	}
+
+	// Fetch the data transfer cost per GB (if source and dest regions are the same, this will be 0).
+	dataTransferCostPerGB, err := s.priceFetcher.GetDataTransferCostPerGB(sourceRegion, destRegion)
+	if err != nil {
+		return nil, fmt.Errorf("failed to get data transfer cost per GB: %v", err)
+	}
+
+	// Fetch the monthly cost of the volume that was created from the snapshot.
+	volumeFromSnapshotMonthlyCost, err := s.priceFetcher.GetVolumeMonthlyCostPerGB(ctx, destRegion, fromSnapshotVolumeType)
+	if err != nil {
+		return nil, fmt.Errorf("failed to get volume monthly cost per GB: %v", err)
+	}
+
+	// 0 in case of same region
+	dataTransferCost := dataTransferCostPerGB * scanSizeGB
+	// 0 in case of same region
+	sourceSnapshotCost := sourceSnapshotMonthlyCost * ((float64(jobCreationTimeSec + scanDurationSec)) / SecondsInAMonth) * scanSizeGB
+
+	destSnapshotCost := destSnapshotMonthlyCost * ((float64(jobCreationTimeSec + scanDurationSec)) / SecondsInAMonth) * scanSizeGB
+	volumeFromSnapshotCost := volumeFromSnapshotMonthlyCost * ((float64(jobCreationTimeSec + scanDurationSec)) / SecondsInAMonth) * scanSizeGB
+	scannerCost := scannerPerHourCost * ((float64(jobCreationTimeSec + scanDurationSec)) / SecondsInAnHour)
+	scannerRootVolumeCost := scannerRootVolumeMonthlyCost * ((float64(jobCreationTimeSec + scanDurationSec)) / SecondsInAMonth) * float64(scannerRootVolumeSizeGB)
+
+	jobTotalCost := sourceSnapshotCost + volumeFromSnapshotCost + scannerCost + scannerRootVolumeCost + dataTransferCost + destSnapshotCost
+
+	// Create the Estimation object base on the calculated data.
+	estimation := models.Estimation{
+		Cost: utils.PointerTo(float32(jobTotalCost)),
+		CostBreakdown: &[]models.CostBreakdownComponent{
+			{
+				Cost:      float32(sourceSnapshotCost),
+				Operation: string(SourceSnapshot),
+			},
+			{
+				Cost:      float32(destSnapshotCost),
+				Operation: string(DestinationSnapshot),
+			},
```

Comment:
I'm not sure this should be in the breakdown if its not something that is part of the operation for this asset right? Same with dataTransferCost, if we don't need to do dataTransfer it shouldn't be listed even if its "0".